src/main/java/uk/gov/justice/digital/hmpps/whereabouts/services/CourtService.kt

```diff
@@ -62,4 +72,67 @@ class CourtService(
         )
       }.toSet()
   }
+
+  @Transactional
+  fun createVideoLinkBooking(specification: VideoLinkBookingSpecification): Long {
+    val mainId = savePrisonAppointment(specification, specification.main)
+    val preId = specification.pre?.let { savePrisonAppointment(specification, it) }
+    val postId = specification.post?.let { savePrisonAppointment(specification, it) }
+
+    val persistentBooking = videoLinkBookingRepository.save(
+      VideoLinkBooking(
+        pre = preId?.let { toAppointment(preId, HearingType.PRE, specification) },
+        main = toAppointment(mainId, HearingType.MAIN, specification),
+        post = postId?.let { toAppointment(postId, HearingType.POST, specification) }
+      )
+    )
+    trackVideoLinkBookingCreated(persistentBooking)
+    return persistentBooking.id!!
+  }
+
+  private fun trackVideoLinkBookingCreated(booking: VideoLinkBooking) {
+    val properties = mutableMapOf(
+      "id" to (booking.id?.toString()),
+      "bookingId" to booking.main.bookingId.toString(),
+      "court" to booking.main.court,
+      "user" to authenticationFacade.currentUsername,
```

Comment:
shame we can't add the prison... but we don't know the prison here so that's fine.